source/fundamentals/authentication.txt

```diff
@@ -1 +1,208 @@
+on establishing a connection to your MongoDB cluster, see the
+:ref:`csharp-connection` guide.
+
+Specify an Authentication Mechanism
+-----------------------------------
+
+You can specify your authentication mechanism and credentials when connecting to
+MongoDB using either of the following:
+
+- A connection string
+- A ``MongoCredential`` factory method
+
+A **connection string**, also known as a **connection uri**, is a string that
+tells the driver how to connect to a MongoDB deployment and how to behave while
+connected.
+
+The ``MongoCredential`` class contains factory methods for each of the supported
+authentication mechanisms. 
+
+Mechanisms
+----------
+
+The following examples specify authentication mechanisms using the following
+placeholders:
+
+- ``username`` - Your MongoDB username
+- ``password`` - Your MongoDB user's password
+- ``hostname`` - The network address of your MongoDB server, accessible by your client
+- ``port`` - The port number of your MongoDB server
+- ``authenticationDb`` - The MongoDB database that contains your user's authentication
+  data. If you omit this parameter, the driver uses the default value ``admin``.
+
+Select the :guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying your authentication mechanism.
+
+.. _csharp-authentication-default:
+
+Default
+~~~~~~~
+
+The default authentication mechanism setting uses one of the following
+authentication mechanisms, depending on what MongoDB versions your server supports:
+
+- ``SCRAM-SHA-256``
+- ``SCRAM-SHA-1``
+- ``MONGODB-CR``
+
+.. note::
+
+   MongoDB version 4.0 uses SCRAM as the default mechanism, and no longer
+   supports ``MONGODB-CR``. These versions automatically determine which
+   version of SCRAM to use.
+
+
+Select the :guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax for specifying the default authentication mechanism:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: default-connection-string
+
+      .. code-block:: csharp
+
+         var mongoClient = new MongoClient("mongodb://<username>:<password>@<hostname>:<port>/?authSource=<authenticationDb>");
+
+   .. tab:: MongoCredential
+      :tabid: default-mongo-credential
+
+      .. code-block:: csharp
+
+         var credentials = MongoCredential.CreateCredential("<authenticationDb>", "<username>", "<password>");
+         var settings = MongoClientSettings.FromConnectionString("<host>");
+         settings.Credential = credentials;
+         var mongoClient = new MongoClient(settings);
+
+.. _csharp-scram-sha-256:
+
+SCRAM-SHA-256
+~~~~~~~~~~~~~
+
+``SCRAM-SHA-256`` is a salted challenge-response authentication mechanism (SCRAM)
+that uses your username and password, encrypted with the ``SHA-256`` algorithm,
+to authenticate your user. 
+
+.. tip:: Default Mechanism
+
+   MongoDB version 4.0 and later uses SCRAM as the default authentication mechanism.
```

Comment:
which SCRAM method? It's a little confusing that this same tip is in both this section and the next.

Reply:
I updated the tip to make it more clear.